tf2_ros_py/tf2_ros/buffer_client.py

```diff
@@ -79,7 +80,8 @@ def __init__(
         """
         tf2_ros.BufferInterface.__init__(self)
         self.node = node
-        self.action_client = ActionClient(node, LookupTransform, action_name=ns)
+        # self.action_client = ActionClient(node, LookupTransform, action_name=ns)
+        self.service_client = node.create_client(LookupTransform, ns, callback_group=MutuallyExclusiveCallbackGroup())
```

Comment:
Many nodes that will contain a Buffer client will need to have a MultiThreadedExecutor to avoid deadlocks. Most commonly a deadlock can happen if the buffer client is called within a subscriber or timer callback.

Reply:
why can the client only process one request at a time? Not sure if needed, but one might e.g. want to get the location of the pallet from left and right depth camera simultaneously

Reply:
> why can the client only process one request at a time?

You mean why is it MutuallyExclusiveCallbackGroup and not Reentrant? My goal with that was only to put the client in a different group than any of the timers/subscribers/... created by the node using a buffer client.

I didn't think about the difference between MutuallyExclusiveCallbackGroup and Reentrant here :thinking: I guess like this, in the scenario where we have 2 threads making a service call, they would block each other? I don't mind setting it to Reentrant, I don't see a risk here.

 > Not sure if needed, but one might e.g. want to get the location of the pallet from left and right depth camera simultaneously
 
But in this case I think the simplest would just to do them consecutively in one thread but should pass the same "time.now()" that is created before calling the first service

Reply:
I'm not worried about a deadlock. I could only see a performance benefit in some (possibly obscure) cases: when trying to use the buffer_client in a parallelized way. However we haven't done this so far for the listener. So I think the current callback group is also fine.